### PR TITLE
ATO-402- Add: class AliElasticSearchRoot 

### PR DESCRIPTION
**Pull request by miranov25**
## ATO-402- Add: class AliElasticSearchRoot

* AliElasticSearchRoot - C++ interface to the Elastic --> root tree like-   using gSystem->GetFromPipe, Exec, Elastic, painless scripting,  curl,  jq
* Originally planned usage of  higher level C++ interface not possible for a moment
  * C++11 or better C++14 and ROOT6 needed (e.g https://github.com/QHedgeTech/cpp-elasticsearch/tree/master/src )
  * Using older  C++  and Root 5 to represent hierarchical structure root trees to be used
  * Query language similar to TTree::Draw  - suing elastic painless scripting language (see query examples)

### Example usage
```c++
hostName="localhost:9200";            // local elastic
 // or remote ElastichostName="https://localhost:9206";    //  connect to remote via tunnel e.g ssh mivanov@lxplus.cern.ch -f -N -L 9206:es-alice.cern.ch:9203
 //
 AliElasticSearchRoot elastic(hostName,kFALSE,1);
 elastic.SetCertificate();           // default certification="--insecure --cacert CERNRootCertificationAuthority2.crt --netrc-file $HOME/.globus/.elastic/.netrc"
 elastic.ls();                       // list all indeces
 elastic.ls("alice");                // list incdeces containing alice
 elastic.ls(".alice_mc");             // list indeces containing alice_mc
 elastic.Print();
 // GetIndexLayeout
 elastic.GetIndexLayout("/alice/logbook_test0/")->Print();
 elastic.GetIndexLayout("/alice_mc/passguess/")->Print();
 // make a Tree like query
 TString select="";
 select=  elastic.select("/alice/qatpc_test0/","run:meanMIP:meanTPCncl","run>240000&&run<246844&&meanMIP>50",0,1000,"jqEQueryStat,jqEQueryHeader")
 select=  elastic.select("/alice/logbook_test0/","run:totalSubEvents:ctpDuration","run>140000",0,10000,"jqEQueryStat,jqEQueryHeader")
 select=  elastic.select("/alice_mc/passguess/","*","1",0,1000,"jqEQueryStat,jqEQueryHeader,jqEQueryArrayNamed");  // get list of all runs output in query.json
 //
 select=  elastic.select("/alice/qatpc_test0/","*","run>240000&&run<246844&&meanMIP>50",0,1000,0)
```

### CODE REVIEW

**Comment by miranov25** [comment]
@shahor02, @dberzano, @pzhristov 
Can you approve request?  I would like to make AliElastic interface available with qa-tools community.
For some reason I was not able to approve the request.

**Comment by shahor02** [comment]
+1

On 9 Jun 2017 11:24 am, miranov25 <notifications@github.com> wrote:

@shahor02<https://github.com/shahor02>, @dberzano<https://github.com/dberzano>, @pzhristov<https://github.com/pzhristov>
Can you approve request? I would like to make AliElastic interface available with qa-tools community.
For some reason I was not able to approve the request.

File: STAT/AliElasticSearchRoot.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/// \author marian  Ivanov marian.ivanov@cern.ch
+
+#include "TObject.h"
+#include <map>
+#include <string>
+#include "TPRegexp.h"
+
+
+class AliElasticSearchRoot : public TObject{
+
```
Comment (sawenzel) [inline STAT/AliElasticSearchRoot.h]: Is there any particular reason why this inherits from TObject?

Comment (miranov25) [inline STAT/AliElasticSearchRoot.h]: No a particular reason. 
Except of fact I use .ls(), Print()

Comment (miranov25) [inline STAT/AliElasticSearchRoot.h]: @sawenzel  In case you have time we can meet on Vidyo to discuss current implementation (not full) and further plans

**Pull request merged**